ImperatorToCK3/Mappers/Localizaton/LocalizationMapper.cs

```diff
@@ -0,0 +1,186 @@
+            if (!Directory.Exists(languagePath)) {
+                return;
+            }
+            var fileNames = SystemUtils.GetAllFilesInFolderRecursive(languagePath);
+            foreach (var fileName in fileNames) {
+                var filePath = Path.Combine(languagePath, fileName);
+                try {
+                    var stream = File.OpenText(filePath);
+                    var reader = new BufferedReader(stream);
+                    ScrapeStream(reader, language);
+                    stream.Close();
+                } catch (Exception e) {
+                    Logger.Log(LogLevel.Warning, "Could not parse localization file " + filePath + " : " + e);
+                }
+            }
+        }
+        public void ScrapeStream(BufferedReader reader, string language) {
+            while (!reader.EndOfStream) {
+                var (key, loc) = DetermineKeyLocalizationPair(reader.ReadLine());
+                if (key is null || loc is null) {
+                    continue;
+                }
+
+                var gotValue = localizations.TryGetValue(key, out var locBlock);
+                if (gotValue) {
+                    locBlock.SetLocForLanguage(language, loc);
+                } else {
+                    var newBlock = new LocBlock();
+                    newBlock.SetLocForLanguage(language, loc);
+                    localizations.Add(key, newBlock);
+                }
+            }
+        }
+        private static KeyValuePair<string?, string?> DetermineKeyLocalizationPair(string? line) {
+            if (line == null || line.Length < 4 || line[0] == '#' || line[1] == '#') {
+                return new KeyValuePair<string?, string?>();
+            }
+            var sepLoc = line.IndexOf(':');
+            if (sepLoc == -1) {
+                return new KeyValuePair<string?, string?>();
+            }
+            var key = line.Substring(1, sepLoc - 1);
+            var newLine = line.Substring(sepLoc + 1);
+            var quoteLoc = newLine.IndexOf('\"');
+            var quote2Loc = newLine.LastIndexOf('\"');
+            if (quoteLoc == -1 || quote2Loc == -1 || quote2Loc - quoteLoc == 0) {
+                return new KeyValuePair<string?, string?>(key, null);
+            }
+            var value = newLine.Substring(quoteLoc + 1, quote2Loc - quoteLoc - 1);
+            return new KeyValuePair<string?, string?>(key, value);
+        }
+        public LocBlock? GetLocBlockForKey(string key) {
+            var gotValue = localizations.TryGetValue(key, out var locBlock);
+            if (!gotValue) {
+                return null;
+            }
+
+            if (!string.IsNullOrEmpty(locBlock.english) &&
+                (string.IsNullOrEmpty(locBlock.french) ||
+                string.IsNullOrEmpty(locBlock.german) ||
+                string.IsNullOrEmpty(locBlock.russian) ||
+                string.IsNullOrEmpty(locBlock.simp_chinese) ||
+                string.IsNullOrEmpty(locBlock.spanish))
```

Comment:
It'd be worth creating a small function that returns a bool and contains all this logic.